client/state/ui/layout-focus/reducer.js

```
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+
+const initialState = { current: null, previous: null, next: null };
+const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+const isValidArea = area => includes( validAreas, area );
+
+export default function layoutFocus( state = initialState, action ) {
+	switch ( action.type ) {
+		case LAYOUT_FOCUS_SET:
+			if ( action.area === state.current || ! isValidArea( action.area ) ) {
```

Thinking on the `isValidArea`, should we be so proactive about validating? Thinking instead it's more valuable to fail loudly. Ideally (and probably for a subsequent pull request) we might extract these section names out into an object of constants.